### DISCUSSION TITLE
Update question #232

### DISCUSSION THREAD

**Discussion by Samseg99** [xcompact3d/Incompact3d#234] [Q&A] [answered]
Dear all, 
We have made progress with respect to post #232 the current state is: we are trying to run a Plane Couette flow (see Papavassiliou,  Hanratty, IJHFF, Volume 18, Issue 1, 1997) using Incompact3d.  We have decided for a symmetric arrangement, i.e. the lower wall moving at -U/2, the upper wall at +U/2.  

In this context we have commented the "constant flow rate" section in  BC-Channel-flow.f90 like that: 

!if (.not. cpg ) then ! if not constant pressure gradient
!   if (idir_stream == 1) then
!      call channel_cfr(ux,two/three)
!   else
!      call channel_cfr(uz,two/three)
!   endif
!end if

but after the flow becomes statistically steady, the mean velocity profile close to the two walls is unexpectedly larger than the wall velocity |ux| > 0.5 and behaves irregular, with one overshoot at the first computational point in the fluid. The solution is symmetric though. See the mean velocity profile attached.

![average_umean_plot](https://github.com/xcompact3d/Incompact3d/assets/118732913/12eca968-4caf-4e9d-a7d5-19064185fee1)
![U+](https://github.com/xcompact3d/Incompact3d/assets/118732913/bbfaafb3-27c5-45eb-a43e-b25727cccc58)

**Comment by slaizet** [comment, marked as answer]
Hi,

I am afraid I am not able to reproduce your problem. I just ran a quick simulation on a coarse mesh, and the picture below is the velocity field ux once the flow is fully established (statistics are also fine). You can see from the picture that ux is staying between -0.5 and 0.5. I have no issues at the top/bottom boundaries. I did the same as you, I just commented the forcing term from the channel flow, imposed 0.5 at the top and -0.5 at the bottom. I will further investigate, just in case.

Best,

Sylvain

![Screenshot from 2023-10-23 16-24-22](https://github.com/xcompact3d/Incompact3d/assets/45965875/18d3e598-28ae-4167-8005-f188ab01f74c)

**Reply by Samseg99** [reply]
Thanks for the answer
Can I ask the parameter you use in the simulation? If the changing in the code are the same I think that I have not consider something in the parameter file "input.i3d".
Last evening I tried to simulate with a coarse mesh and I have the same problem.

This is the Input file I used last evening:

`! -*- mode: f90 -*-

!===================
&BasicParam
!===================

! Flow type (1=Lock-exchange, 2=TGV, 3=Channel, 4=Periodic hill, 5=Cylinder, 6=dbg-schemes)
itype = 3

! Domain decomposition
p_row=0               ! Row partition
p_col=0               ! Column partition

! Mesh
nx=64                ! X-direction nodes
ny=33                 ! Y-direction nodes
nz=32                 ! Z-direction nodes
istret = 2            ! y mesh refinement (0:no, 1:center, 2:both sides, 3:bottom)
beta = 0.259065151    ! Refinement parameter (beta) 0.259065151

! Domain
xlx = 4.           ! Lx (Size of the box in x-direction)
yly = 2.              ! Ly (Size of the box in y-direction)
zlz = 2.              ! Lz (Size of the box in z-direction)

! Flow parameters
re  = 2660.           ! nu=1/re (Kinematic Viscosity)
cpg = F               ! if cpg=T, then re is friction Reynolds number

! Time stepping
dt = 0.001            ! Time step
ifirst = 1            ! First iteration
ilast = 1000000        ! Last iteration

! Enable modelling tools
ilesmod=0             ! if 0 then DNS
numscalar = 0         ! How many scalars? (Set to zero to disable scalars)
iibm=0                ! Flag for immersed boundary method

! Boundary and initial conditions
iin = 1               ! Inflow conditions (1: classic, 2: turbinit, 4: SEM)
u1  = 1.              ! u1 (max velocity) (for inflow condition)
u2  = 1.              ! u2 (min velocity) (for inflow condition)
init_noise = 0.5    ! Turbulence intensity (1=100%) !! Initial condition
inflow_noise = 0.0    ! Turbulence intensity (1=100%) !! Inflow condition
idir_stream  = 1      ! Index of the streamwise direction (1=X, 3=Z, 2 not available)

nclx1 = 0
nclxn = 0
ncly1 = 2
nclyn = 2
nclz1 = 0
nclzn = 0

/End

!====================
&NumOptions
!====================

! Spatial derivatives
ifirstder = 4         ! (1->2nd central, 2->4th central, 3->4th compact, 4-> 6th compact)
isecondder = 5        ! (1->2nd central, 2->4th central, 3->4th compact, 4-> 6th compact, 5->hyperviscous 6th)
ipinter = 3           ! interpolation scheme (1: classic, 2: optimized, 3: optimized agressive)

! Time scheme
iimplicit = 2
itimescheme = 3       ! Time integration scheme (1->Euler,2->AB2, 3->AB3, 4->AB4,5->RK3,6->RK4, 7-->CN2+AB3)

! Dissipation control
nu0nu = 4.0             ! Ratio between hyperviscosity/viscosity at nu
cnu = 0.44               ! Ratio between hypervisvosity at k_m=2/3pi and k_c= pi

/End

!=================
&InOutParam
!=================

! Basic I/O
irestart = 0          ! Read initial flow field ?
icheckpoint = 5000    ! Frequency for writing backup file
ioutput = 10000       ! Frequency for visualization
ilist = 25            ! Frequency for writing to screen        
nvisu = 1             ! Size for visualisation collection
nprobes = 4           ! Number of probes

/End

!=================
&Statistics
!=================

wrotation = 0.25      ! rotation speed to trigger turbulence
spinup_time = 10000    ! number of time steps with a rotation to trigger turbulence
nstat = 1             ! Size arrays for statistic collection
initstat = 40000      ! Time steps after which statistics are collected

/End

!=================
&ProbesParam
!=================

flag_all_digits = T     ! By default, only 6 digits, 16 when True
flag_extra_probes = T   ! By default, gradients are not monitored
xyzprobes(1,1) = 0.1    ! X position of probe 1
xyzprobes(2,1) = 0.1    ! Y position of probe 1
xyzprobes(3,1) = 0.1    ! Z position of probe 1
xyzprobes(1,2) = 0.1    ! X position of probe 2
xyzprobes(2,2) = 0.2    ! Y position of probe 2
xyzprobes(3,2) = 0.1    ! Z position of probe 2
xyzprobes(1,3) = 0.1    ! X position of probe 3
xyzprobes(2,3) = 0.3    ! Y position of probe 3
xyzprobes(3,3) = 0.1    ! Z position of probe 3
xyzprobes(1,4) = 0.1    ! X position of probe 4
xyzprobes(2,4) = 0.4    ! Y position of probe 4
xyzprobes(3,4) = 0.1    ! Z position of probe 4

/End

!########################
! OPTIONAL PARAMETERS
!#######################

!================
&ScalarParam
!================

Sc(1) = 1.0           ! Schmidt number

nclxS1 = 0
nclxSn = 0
nclyS1 = 2
nclySn = 2
nclzS1 = 0
nclzSn = 0

/End

!================
&LESModel
!================

iles = 0             ! LES Model (1: Phys Smag, 2: Phys WALE, 3: Phys dyn. Smag, 4: iSVV, 5: dyn SEV)
smagcst = 0.14        ! Smagorinsky constant
SmagWallDamp = 0      ! 1: Mason and Thomson Damping function, otherwise OFF
walecst = 0.5         ! WALES Model Coefficient
iconserv = 0          ! Formulation SGS divergence (0: non conservative, 1: conservative)

/End

&CASE
/End`

**Comment by slaizet** [comment]
Can you try with an explicit time advancement scheme?
```
! Time scheme
iimplicit = 0
itimescheme = 5 
```
You may have not updated the BC properly with the implicit time advancement strategy.

Cheers,

Sylvain

**Reply by Samseg99** [reply]
I've started a simulation with explicit time advancement and it seems to be working.
I'll analyze the implicit time advancement strategy better for the future simulations.

Thanks so much

Samuele
